libmuscle/cpp/src/libmuscle/tests/test_api_guard.cpp

```diff
@@ -0,0 +1,211 @@
+#include <gtest/gtest.h>
+
+#include <libmuscle/api_guard.hpp>
+
+#include <vector>
+#include <set>
+
+using libmuscle::impl::APIGuard;
+using libmuscle::impl::APIPhase;
+
+int main(int argc, char *argv[]) {
+    ::testing::InitGoogleTest(&argc, argv);
+    return RUN_ALL_TESTS();
+}
+
+TEST(libmuscle_api_guard, test_no_checkpointing_support) {
+    auto guard = APIGuard(false, true);
```

Comment:
I'd expect copy/move elision to kick in here indeed, though it is only required by the C++ standard since C++17 <https://en.cppreference.com/w/cpp/language/copy_elision>
```
T x = T(T(f())); // only one call to default constructor of T, to initialize x
```

I don't think the `auto` keyword changes anything to that.

I'll keep it like this, it's only a test anyway :)

Reply:
:+1: